### DISCUSSION TITLE
Authorize to realm with OAuth2 token over keycloak api

### DISCUSSION THREAD

**Discussion by HerrDerb** [keycloak/keycloak#32966] [Q&A]
I am trying to get access to my realm by providing my oauth2 bearer token.

We have a open id connect service configured for our keycloak. 
I would like to authorize to my realm using my bearer token. 
I tried GET https://my-host/auth/realms/my-realm/protocol/openid-connect/token with my bearer token in the headers, which did not work (405)

What api call do I need to make to authorize to my realm using my oauth2 bearer token?
(I hope I did not mix up things)

**Comment by darius-m** [comment]
What would authenticating to your realm imply? What are you trying to get in the end?

A bearer token is an authentication token in itself, you are already authenticated in the realm (more specifically you have authenticated to a client in the realm)  if you have it. Now, what you can do with it, is a different story - tokens should, by default, have an "audience" set for them, so the token released for a client cannot be used to authenticate as the user in a different service. This is a security measure, so clients that use bearer tokens (common in REST APIs) know they should not trust all tokens, even if they are released by a trusted SSO service, in case some of the other clients are rogue or compromised.

**Comment by HerrDerb** [comment]
I close this question again, as I am not sure how to ask at the moment
